research/picai/fl_nnunet/nnunet_client.py

```diff
@@ -542,13 +559,44 @@ def empty_cache(self) -> None:
         else:
             pass
 
+    def update_before_train(self, current_server_round: int) -> None:
+        """
+        Reset LR at beginning of training so that initial log str has correct LR and
+        not the LR from the previous round
+        """
+        self.nnunet_trainer.lr_scheduler.step(0)
+
+    def update_before_step(self, step: int) -> None:
+        """
+        Update the learning rate. Need to define this in case train by steps is being
+        used. This is the current step for the entire round, not the current epoch.
+        """
+        # By default nnunet lr schedulers use epochs
```

Comment:
We have this comment above 
```
Default nnunet behaviour is do decrease LR every 250 steps
```
but here we're saying that the LR schedulers use epochs by default, which seems contradictory. However, it's possible I'm missing a nuance here?

Reply:
There is a nuance and I've clarified in the comments. Default nnunet behaviour is to define 1 epoch as 250 steps, train for "1000" epochs and decrease the learning rate every "epoch". My current implementation defines an epoch as what it actually is ((num_samples / batch_size) steps), so we are technically doing it differently right now. My next PR after this one is merged is going to address this by updating based on the number of steps instead of the number of epochs. Additionally I'm considering making a LrScheduler base class that takes the server round, current step, current epoch etc. as arguments when taking a step, as well as some subclasses that will just wrap easily around pytorch lr schedulers. This will be in the next PR